app/portainer/views/home/homeController.js

```diff
@@ -1,95 +1,132 @@
+            }
+            if (status === 1) {
+              if (endpoint.Type === 3) {
+                switchToAzureEndpoint(endpoint);
+              } else {
+                switchToDockerEndpoint(endpoint);
+              }
+            } else {
+              if (endpoint.Type === 3) {
+                Notifications.error('Failure', '', 'Endpoint is unreachable. Offline browsing disabled for Azure endpoints.');
+              } else if (endpoint.Snapshots[0] && endpoint.Snapshots[0].Swarm === true) {
+                Notifications.error('Failure', '', 'Endpoint is unreachable. Connect to another swarm manager.');
+              } else if (!endpoint.Snapshots[0]) {
+                Notifications.error('Failure', '', 'Endpoint is unreachable and there is no snapshot available for offline browsing.');
+              } else {
+                switchToDockerEndpoint(endpoint);
+              }
+            }
+          });
+      };
 
-  $scope.dismissImportantInformation = function(hash) {
-    StateManager.dismissImportantInformation(hash);
-  };
+      $scope.dismissImportantInformation = function (hash) {
+        StateManager.dismissImportantInformation(hash);
+      };
 
-  $scope.dismissInformationPanel = function(id) {
-    StateManager.dismissInformationPanel(id);
-  };
+      $scope.dismissInformationPanel = function (id) {
+        StateManager.dismissInformationPanel(id);
+      };
 
-  function triggerSnapshot() {
-    EndpointService.snapshot()
-    .then(function success() {
-      Notifications.success('Success', 'Endpoints updated');
-      $state.reload();
-    })
-    .catch(function error(err) {
-      Notifications.error('Failure', err, 'An error occured during endpoint snapshot');
-    });
-  }
+      function triggerSnapshot() {
+        EndpointService.snapshot()
+          .then(function success() {
+            Notifications.success('Success', 'Endpoints updated');
+            $state.reload();
+          })
+          .catch(function error(err) {
+            Notifications.error('Failure', err, 'An error occured during endpoint snapshot');
+          });
+      }
 
-  $scope.triggerSnapshot = function() {
-    ModalService.confirmEndpointSnapshot(function (result) {
-      if(!result) { return; }
-      triggerSnapshot();
-    });
-  };
+      $scope.triggerSnapshot = function () {
+        ModalService.confirmEndpointSnapshot(function (result) {
+          if (!result) {
+            return;
+          }
+          triggerSnapshot();
+        });
+      };
 
-  function switchToAzureEndpoint(endpoint) {
-    StateManager.updateEndpointState(endpoint.Name, endpoint.Type, [])
-    .then(function success() {
-      $state.go('azure.dashboard');
-    })
-    .catch(function error(err) {
-      Notifications.error('Failure', err, 'Unable to connect to the Azure endpoint');
-    });
-  }
+      function switchToAzureEndpoint(endpoint) {
+        EndpointProvider.setEndpointID(endpoint.Id);
```

Comment:
I think endpoint provider should have `.setEnpoint` function, I guess this happens anytime on changing endpoint

Reply:
IMO should be part of a specific refactor that only targets the state manager. Related to endpoint provider refactor as well.